packages/backend/src/tokens/keys.ts

```diff
@@ -202,6 +208,31 @@ async function fetchJWKSFromBAPI(apiUrl: string, key: string, apiVersion: string
   });
 
   if (!response.ok) {
+    const errors = await response.json().then(({ errors }) => {
+      if (!errors) {
+        return false;
+      }
+
+      const invalidSecretKey = getErrorObjectByCode(errors, TokenVerificationErrorCode.InvalidSecretKey);
+
+      if (!invalidSecretKey) {
+        return false;
+      }
+
+      return {
+        ...invalidSecretKey,
+        reason: TokenVerificationErrorReason.InvalidSecretKey,
+      };
+    });
+
+    if (errors) {
```

Comment:
🙃 `errors` suggests an array of errors. I would expect either a singular `error` object or a plural `errors` array of objects. Eg
```typescript
const json = await response.json();
const errors = buildErrorOrSomething(json?.errors);

 if (hasInvalidSecretKeyError(errors)) {
    const message = getErrorObjectByCode(errors, TokenVerificationErrorCode.InvalidSecretKey);
    const reason = TokenVerificationErrorReason.InvalidSecretKey;

    throw new TokenVerificationError({
        action: TokenVerificationErrorAction.ContactSupport,
        message,
        reason,
      });
 }